### ISSUE TITLE
Document newline convention required by twisted.mail.pop3.IMailbox.getMessage

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2072]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2072|
|Type|enhancement|
|Created|2006-09-10 15:12:33Z|

The POP3 server will perform a transformation on the bytes read from the file returned by getMessage() which will corrupt any message which does not use \n to delimit lines.  This should be documented in the interface.

<details><summary>Searchable metadata</summary>

```
trac-id__2072 2072
type__enhancement enhancement
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__documentation documentation
time__1157901153000000 1157901153000000
changetime__1164902472000000 1164902472000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun removed owner|
|-|-|

Perhaps this would suffice:

```
Index: twisted/mail/pop3.py
===================================================================
--- twisted/mail/pop3.py	(revision 18490)
+++ twisted/mail/pop3.py	(working copy)
@@ -895,6 +895,8 @@
         @param index: The number of the message to retrieve
 
         @rtype: A file-like object
+        @return: A file containing the message data with lines delimited by
+        C{\n}.
         """
 
     def getUidl(index):
```

But is any other information missing from the interface?

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_2"></a>|@jml set owner to @exarkun|
|-|-|

I'm not looking hard enough to find anything as subtle as the original bug. However, `undeleteMessages` has a confusing and grammatically incorrect docstring.

The main confusion is wrt its return value. The docstring says the method undeletes message_s_, but then goes on to speak of the return value as singular.

It would be nice if you added some tests that explicitly test IMailBox, that way there would be an easier way to verify the docstrings.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun removed owner|
|-|-|

Committed the above patch and a modification to undeleteMessages' docstring to mailbox-interface-clarification-2072.  I am comfortable with the current level of test coverage for newline-handling in the POP3 server (it is covered pretty explicitly by testRETR of test_pop3.py.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to PenguinOfDoom|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)<a name="note_5"></a>|@jerub set owner to @exarkun|
|-|-|

Perfect! thankyou for your attention to detail. Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set status to `closed`|
|-|-|

(In [18792]) Merge mailbox-interface-clarification-2072

Author: exarkun
Reviewer: jerub
Fixes [#2072](https://github.com/twisted/twisted/issues/2072)

Clarify the docstrings of two methods of pop3.IMailbox: explain the newline
convention expected by the POP3 server from application code and fix the
wording in undeleteMessages so that it makes sense.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
